### ISSUE TITLE
Publishing custom artifacts that are built on multiple hosts

### ISSUE THREAD

**Issue by Him188** [vanniktech/gradle-maven-publish-plugin#889]
Hi, thanks for your hard work!

TL;DR: 

My request is, to add a `EmptyPlatform` or `CustomPlatform` where the plugin does not configure anything (e.g. components), but let myself setup the only artifacts needed. 

----

Full story:

I'm trying to publish a Java library with JNI natives to Maven Central. It should support Android and desktop JVM ( mac/win/linux), so there will be `.dll`, `.so` files for each of the platfrom. Native libraries can only be built on the correct host OS, for example, `.dylib` can only be built on macOS.

I'm planning to publish the natives with classifiers:

```kotlin
mavenPublishing {
    configure(JavaLibrary(JavadocJar.Empty(), true))
    publishToMavenCentral(SonatypeHost.CENTRAL_PORTAL)
    signAllPublications()
    // pom omitted
}

val nativeJar = tasks.register("nativeJar", Jar::class.java) {
    dependsOn(copyNativeFiles)
    group = "anitorrent"
    archiveClassifier.set(getOsTriple())
    from(copyNativeFiles.map { it.outputs.files.singleFile })
}

publishing {
    publications {
        getByName("maven", MavenPublication::class) {
            artifact(nativeJar)
        }
    }
}
```

The above works if I only support one single OS.

But I have to run the publishing for each host OS. Since they are all configured with `JavaLibrary`, they will publish the same `.jar` and `.pom` 3 times, resulting to conflicts on Maven Central. 

Currently I see no option for the plugin to configure an 'EmptyPlatform' where I can publish only the `nativeJar` artifact on some OSes.

-----

I've came up with a possible workround that is, to build the native libraries before any publishing logic. Gather the libraries onto one machine, and publish them together:

```kotlin
publishing {
    publications {
        getByName("maven", MavenPublication::class) {
            artifact(macOSArm64NativeJar)
            artifact(macOSX64NativeJar)
            artifact(windowsX64NativeJar)
            artifact(linuxX64NativeJar)
        }
    }
}
```

I haven't tried this yet. But I think it will be OK.

**Comment by Enaium** [comment]
You are right, I tried your method, and it's worked fine on multiple artifacts.

```kotlin
afterEvaluate {
    publishing {
        publications {
            getByName<MavenPublication>("maven") {
                artifact(source1)
                artifact(source2)
                artifact(source3)
            }
        }
    }
}
```
